istioctl/pkg/auth/upgrader.go

```diff
@@ -0,0 +1,376 @@
+	}
+	return authzPolicies, nil
+}
+
+// convert is the main function that converts RBAC v1alphal1 to v1beta1 for local policy files
+func (ug *Upgrader) convert(authzPolicies *model.AuthorizationPolicies) error {
+	namespaces := authzPolicies.ListNamespacesOfToV1alpha1Policies()
+	if len(namespaces) == 0 {
+		return fmt.Errorf("no namespace found")
+	}
+	// Build a model for each ServiceRole and associated list of ServiceRoleBinding
+	for _, ns := range namespaces {
+		bindingsKeyList := authzPolicies.ListServiceRoleBindings(ns)
+		for _, roleConfig := range authzPolicies.ListServiceRoles(ns) {
+			roleName := roleConfig.Name
+			if bindings, found := bindingsKeyList[roleName]; found {
+				role := roleConfig.Spec.(*rbac_v1alpha1.ServiceRole)
+				m := authz_model.NewModel(role, bindings)
+				err := ug.v1alpha1ModelTov1beta1Policy(m, ns)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// v1alpha1ModelTov1beta1Policy converts the policy of one ServiceRole and a list of associated
+// ServiceRoleBinding to the equivalent AuthorizationPolicy.
+func (ug *Upgrader) v1alpha1ModelTov1beta1Policy(v1alpha1Model *authz_model.Model, namespace string) error {
+	if v1alpha1Model == nil {
+		return fmt.Errorf("v1alpha1 model is nil")
+	}
+	if v1alpha1Model.Permissions == nil || len(v1alpha1Model.Permissions) == 0 {
+		return fmt.Errorf("permissions are empty")
+	}
+	if v1alpha1Model.Principals == nil || len(v1alpha1Model.Principals) == 0 {
+		return fmt.Errorf("principals are empty")
+	}
+	// TODO(pitlv2109): Support more complex cases
+	if len(v1alpha1Model.Permissions) > 1 {
+		return fmt.Errorf("more than one access rule is not supported")
+	}
+	accessRule := v1alpha1Model.Permissions[0]
+	operations := convertAccessRuleToOperation(&accessRule)
+	if operations == nil {
+		return fmt.Errorf("cannot convert access rule to operation")
+	}
+	sources := convertBindingToSources(v1alpha1Model.Principals)
+	// If there is no services field or it's defined as "*", it's equivalent to an AuthorizationPolicy
+	// with no workload selector
+	authzPolicyConfig := model.Config{
+		ConfigMeta: model.ConfigMeta{
+			Type:      schemas.AuthorizationPolicy.Type,
+			Namespace: namespace,
+		},
+		Spec: &rbac_v1beta1.AuthorizationPolicy{
+			Selector: &v1beta1.WorkloadSelector{},
+			Rules: []*rbac_v1beta1.Rule{
+				{
+					From: sources,
+					To: []*rbac_v1beta1.Rule_To{
+						{
+							Operation: operations,
+						},
+					},
+				},
+			},
+		},
+	}
+	if len(accessRule.Services) == 0 || accessRule.Services[0] == "*" {
+		authzPolicyConfig.Name = fmt.Sprintf("%s-services", namespace)
+		authzPolicyConfig.Spec.(*rbac_v1beta1.AuthorizationPolicy).Selector = nil
+		ug.AuthorizationPolicies = append(ug.AuthorizationPolicies, authzPolicyConfig)
+		return nil
+	}
+	for _, service := range accessRule.Services {
+		serviceName := strings.Split(service, ".")[0]
+		authzPolicyConfig.Name = fmt.Sprintf("%s-%s", serviceName, namespace)
```

Comment:
You have "namespace" in the field before "-". Let's be consistent with the naming scheme, i.e., namespace-serviceName.

Reply:
Done. Thanks!